plugins/module_utils/network/iosxr/config/bgp_global/bgp_global.py

```diff
@@ -0,0 +1,421 @@
+            module=module,
+            resource="bgp_global",
+            tmplt=Bgp_globalTemplate(),
+        )
+        self.parsers = [
+            "router",
+            "bfd_multiplier",
+            "bfd_minimum_interval",
+            "bgp_auto_policy_soft_reset",
+            "bgp_as_path_loopcheck",
+            "bgp_cluster_id",
+            "bgp_default_local_preference",
+            "bgp_enforce_first_as_disable",
+            "bgp_fast_external_fallover_disable",
+            "bgp_install_diversion",
+            "bgp_max_neighbors",
+            "bgp_redistribute_internal",
+            "bgp_router_id",
+            "bgp_scan_time",
+            "bgp_unsafe_ebgp_policy",
+            "bgp_update_delay",
+            "bgp_bestpath_aigp",
+            "bgp_bestpath_as_path_ignore",
+            "bgp_bestpath_as_path_multipath_relax",
+            "bgp_bestpath_med_always",
+            "bgp_bestpath_med_confed",
+            "bgp_bestpath_med_missing_as_worst",
+            "bgp_bestpath_compare_routerid",
+            "bgp_bestpath_cost_community_ignore",
+            "bgp_bestpath_origin_as_use",
+            "bgp_bestpath_origin_as_allow",
+            "bgp_confederation_identifier",
+            "bgp_graceful_restart_set",
+            "bgp_graceful_restart_graceful_reset",
+            "bgp_graceful_restart_restart_time",
+            "bgp_graceful_restart_purge_time",
+            "bgp_graceful_restart_stalepath_time",
+            "bgp_log_message",
+            "bgp_log_neighbor_changes_detail",
+            "bgp_log_neighbor_changes_disable",
+            "bgp_multipath_as_path_ignore_onwards",
+            "bgp_origin_as_validation_disable",
+            "bgp_origin_as_validation_signal_ibgp",
+            "bgp_origin_as_validation_time_off",
+            "bgp_origin_as_validation_time",
+            "bgp_default_information_originate",
+            "bgp_default_metric",
+            "bgp_graceful_maintenance",
+            "ibgp_policy_out_enforce_modifications",
+            "mpls_activate_interface",
+            "mvpn",
+            "nsr_set",
+            "nsr_disable",
+            "socket_receive_buffer_size",
+            "socket_send_buffer_size",
+            "update_in_error_handling_basic_ebgp_disable",
+            "update_in_error_handling_basic_ibgp_disable",
+            "update_in_error_handling_extended_ebgp",
+            "update_in_error_handling_extended_ibgp",
+            "update_out_logging",
+            "update_limit",
+            "rpki_route_value",
+            "rd_auto",
+            "timers_keepalive",
+        ]
+
+    def execute_module(self):
+        """ Execute the module
+
+        :rtype: A dictionary
+        :returns: The result from module execution
+        """
+        if self.state not in ["parsed", "gathered"]:
+            self.generate_commands()
+            self.run_commands()
+        return self.result
+
+    def generate_commands(self):
```

Comment:
We should be failing early and gracefully if BGP is already configured with another ASN.

Reply:
ack